client/src/plugins/user-journey-statistics/event-handlers/__tests__/TabEventHandlerSpec.js

```diff
@@ -0,0 +1,394 @@
+/**
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * Camunda licenses this file to you under the MIT; you may not use this file
+ * except in compliance with the MIT License.
+ */
+
+/* global sinon */
+
+import MixpanelHandler from '../../MixpanelHandler';
+import TabEventHandler from '../TabEventHandler';
+
+import emptyXML from './fixtures/empty.bpmn';
+
+import emptyDMN from './fixtures/empty.dmn';
+
+import brokenForm from './fixtures/broken-form.form';
+
+import engineProfileXML from './fixtures/engine-profile.bpmn';
+
+import engineProfilePlatform from './fixtures/engine-platform.form';
+
+import engineProfileCloud from './fixtures/engine-cloud.form';
+
+import engineProfilePlatformDMN from './fixtures/engine-platform.dmn';
+
+import engineProfileCloudDMN from './fixtures/engine-cloud.dmn';
+
+
+describe('<TabEventHandler>', () => {
+
+  let subscribe, track;
+
+  beforeEach(() => {
+
+    subscribe = sinon.spy();
+
+    track = sinon.spy();
+
+    new TabEventHandler({
+      track,
+      subscribe
+    });
+
+    MixpanelHandler.getInstance().enable('token', 'id', 'stage');
+  });
+
+  it('should subscribe to bpmn.modeler.created', () => {
+
+    // then
+    expect(subscribe.getCall(0).args[0]).to.eql('bpmn.modeler.created');
+  });
+
+
+  it('should subscribe to dmn.modeler.created', () => {
+
+    // then
+    expect(subscribe.getCall(1).args[0]).to.eql('dmn.modeler.created');
+  });
+
+
+  it('should subscribe to form.modeler.created', () => {
+
+    // then
+    expect(subscribe.getCall(2).args[0]).to.eql('form.modeler.created');
+  });
+
+
+  it('should subscribe to tab.closed', () => {
+
+    // then
+    expect(subscribe.getCall(3).args[0]).to.eql('tab.closed');
+  });
+
+
+  it('should send with diagram type: bpmn', async () => {
```

Comment:
I propose to add an integration test to ensure that defaultProps like version and stage will be included

Reply:
I added a test case in MixpanelSpec to test that any event sent has those properties: https://github.com/camunda/camunda-modeler/pull/2941/files#diff-7b30c1c9abd6a969a29bacc7d92e94a0ecdce9dfa8a63c7c08556a21629c0e1dR55